### ISSUE TITLE
Can we try object.toString before Object.prototype.toString?

### ISSUE THREAD

**Issue by arasmussen** [inspect-js/is-equal#78]
Running into the following issue when comparing MongoDB ObjectIds:

```js
import mongoose from 'mongoose';

const foo = mongoose.Types.ObjectId(123);
const bar = mongoose.Types.ObjectId(456);

Object.prototype.toString(foo); // '[Object object]'
foo.toString(); // '000001c81ce7420101e4238e'

Object.keys(foo); // []

isEqual(foo, bar); // true
```

npm versions:
```
mongoose: ^6.12.0
mongodb: ^4.17.0
expect: ^1.20.2
```

expect uses `is-equal: ^1.5.1`, so pushing 1.6.5 would resolve this issue. Happy to submit a PR! Just want to know if it'd be approved.

**Comment by arasmussen** [comment]
To be clear, my proposal is:

https://github.com/inspect-js/is-equal/blob/main/why.js#L46-L50

If `val.toString` exists and is a function, and `other.toString` exists and is a function, call and compare the results. If they differ, specify a why message (that the `toString` results are different).

**Comment by ljharb** [comment]
That’s not a bad comparison to add; we could check valueOf and Symbol.toPrimitive as well, and use es-to-primitive with different hints to cover them all.

That way we don’t have to check specific methods, we can just compare “coercing to a string” and “coercing to a number”, altho we’d have to try/catch to account for things that throw (like symbols), in which case “both throw” would be considered equal.

**Comment by arasmussen** [comment]
@ljharb submitted two PRs, one with the fix, and another for the version bump. Used es-to-primitive with separate hints, as you requested. Lmk if you have feedback.

**Comment by ljharb** [comment]
Thanks, I’ll take a look - versions are never bumped in PRs, though, nor by non-maintainers, anywhere on github.

**Comment by arasmussen** [comment]
Thanks for merging! Sorry to keep bugging you – do you have an ETA on 1.6.5? Eager to get this updated in our project.

**Comment by ljharb** [comment]
It’d be 1.7.0, and not just yet, but hopefully soon.

Fixed by #79.
